lib/macho/load_commands.rb

```diff
@@ -400,6 +400,32 @@ def initialize(view, cmd, cmdsize, segname, vmaddr, vmsize, fileoff,
       @flags = flags
     end
 
+
+    # All sections referenced within this segment.
+    # @return [Array<MachO::Section>] if the Mach-O is 32-bit
+    # @return [Array<MachO::Section64>] if the Mach-O is 64-bit
+    def sections
+      section_klass = case self
+      when MachO::SegmentCommand64
+        MachO::Section64
+      when MachO::SegmentCommand
+        MachO::Section
+      end
+
+      sections = []
+      return sections if nsects.zero?
+
+      offset = view.offset + self.class.bytesize
+
+      nsects.times do
+        section_bin = view.raw_data[offset, section_klass.bytesize]
+        sections << section_klass.new_from_bin(view.endianness, section_bin)
+        offset += section_klass.bytesize
+      end
```

Comment:
How silly of me to not realize we're dealing with a `String` here! Sorry! Not sure how you feel about it, but my suggestion can be rescued with `String#unpack`:

```ruby
sections_bin = view.raw_data[offset, nsects * section_klass.bytesize]
sections_bin.unpack("a#{section_klass.bytesize}" * nsects).map do |section_bin|
  section_klass.new_from_bin(view.endianness, section_bin)
end
```

(I guess I'm equally fine with this version and what we currently have in the PR.)